include/system.h

```diff
@@ -44,7 +44,7 @@ typedef struct
      * @return A pointer to an arbitrary file handle assigned by the filesystem code 
      *         or NULL on error.
      */
-    void *(*open)( char *name, int flags );
+    void *(*open)( const char *name, int flags );
```

Comment:
because I changed `open` from
`int open( char *file, int flags, int mode )`
to
`int open( const char *file, int flags, int mode )`
to match the standard
(including `fcntl.h` (for `O_RDONLY`) revealed that mistake)
and this is called by `open` using that `name`

Reply:
We can't merge this as this breaks backward compatibility with extensive filesystem implementations. You will have to cast your way through this.

Reply:
I wouldn't do that for const correctness honestly